packages/hash/frontend/src/components/GlideGlid/glide-grid.tsx

```diff
@@ -26,21 +27,46 @@ import { uniqueId } from "lodash";
 import { getCellHorizontalPadding } from "./utils";
 import { customGridIcons } from "./utils/custom-grid-icons";
 import { InteractableManager } from "./utils/interactable-manager";
+import { useDrawHeader } from "./utils/use-draw-header";
+import {
+  createHandleHeaderClicked,
+  sortRowData,
+  TableSort,
+} from "./utils/sorting";
+
+export type Row = Record<string, any>;
+export type RowData = Row[];
 
 type GlideGridProps = Omit<
   DataEditorProps,
-  "onColumnResize" | "onColumnResizeEnd" | "onColumnResizeStart" | "columns"
+  | "onColumnResize"
+  | "onColumnResizeEnd"
+  | "onColumnResizeStart"
+  | "columns"
+  | "getCellContent"
+  | "rows"
 > & {
-  resizable?: boolean;
   columns: SizedGridColumn[];
+  rowData: RowData;
+  resizable?: boolean;
+  sortable?: boolean;
+  initialPropertySort?: TableSort<string>;
+  getCellContent: (rowData: any) => (cell: Item) => GridCell;
+  onSort?: (rowData: RowData) => RowData;
```

Comment:
I wasn't expecting `onSort` to change the data after it had been sorted – maybe this can be called something else? I don't have any great ideas for naming though. 

Or could the data be flattened before it's sent into the table and sorted?

Reply:
I'm not sure what I should call it either, what do you think of `rowSortCallback`, `afterSortCallback`, `afterSortRowTransform` or any variation of these?

We could flatten it before it's sent into the table, but that would require changing the sorting logic to sort only the items with `depth: 0` and to move their sub-items with them.

Reply:
I imagine we'll end up with more tables with nested items in it – perhaps the GlideGrid component could be aware of them and handle all the sorting and flattening of nested objects?

Otherwise the next component that needs to send nested things is going to have to implement this `onSort` also

Reply:
I agree `onSort` isn't an ideal name. I think we should have a `sortRows` prop  which does the entire sorting process which also allows you to wrap it with other behaviour, like flatten rows. See my other comment below.

Reply:
Went with @nathggns suggestion in 789280d087f815073f95c354daa52c80ee150bef. 

I think we might want to move the logic of nesting items to `GlideGrid` in the future, but I propose we wait until we actually have to use it more tables.